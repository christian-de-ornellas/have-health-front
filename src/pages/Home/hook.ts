import { useState, useEffect, useRef } from 'react';
import { getClinic } from '../../services/clinic';
import { ClinicResponseOutput } from '../../types/global-types';

export const useHome = () => {
  const isInitialMount = useRef(true);
  const [clinics, setClinics] = useState<ClinicResponseOutput[]>([]);

  const getClinics = async (): Promise<void> => {
    try {
      const response = await getClinic();
      setClinics(response.result);
    } catch (error) {
      return console.error(error);
    }
  };

  useEffect(() => {
    if (isInitialMount.current) {
      isInitialMount.current = false;
    } else {
      getClinics();
    }
  }, []);

  return { clinics };
};
