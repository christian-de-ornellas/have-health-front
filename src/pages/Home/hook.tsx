import { useState, useEffect } from 'react';
import { getClinic } from '../../services/clinic';
import { ClinicResponseOutput } from '../../types/global-types';

export const useHome = () => {
  const [clinics, setClinics] = useState<ClinicResponseOutput[]>([]);

  const fetchClinics = async (): Promise<void> => {
    try {
      const response = await getClinic();
      setClinics(response.result);
    } catch (error) {
      return console.error(error);
    }
  };

  useEffect(() => {
    fetchClinics();
  }, []);

  return { clinics };
};
