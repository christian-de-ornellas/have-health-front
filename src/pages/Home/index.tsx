const Home: React.FC = () => {
  return (
    <>
      <h1>Oi eu sou a home</h1>
    </>
  );
};
export default Home;
