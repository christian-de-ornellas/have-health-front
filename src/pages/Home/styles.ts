import styled from 'styled-components';

export const BlockRegister = styled.div`
padding: 1rem 4rem;
position: absolute;
z-index: 999;
`;
