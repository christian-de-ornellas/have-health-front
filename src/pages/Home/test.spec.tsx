import { render, screen } from '@testing-library/react';
import Home from '.';

describe('<Home />', () => {
  it('should render component', () => {
    render(<Home />);
    expect(
      screen.getByRole('heading', { name: 'Oi eu sou a home' }),
    ).toMatchSnapshot();
  });
});
