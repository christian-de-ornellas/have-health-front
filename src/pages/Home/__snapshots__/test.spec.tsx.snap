// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should  1`] = `
<h1>
  Oi eu sou a home
</h1>
`;

exports[`<Home /> should render and component defined 1`] = `
<h1>
  Oi eu sou a home
</h1>
`;

exports[`<Home /> should render component 1`] = `
<h1>
  Oi eu sou a home
</h1>
`;

exports[`first should  1`] = `
<h1>
  Oi eu sou a home
</h1>
`;
